File: frontend/src/services/classService.js
import api from './axiosConfig';
export const getTrainerClasses = () => api.get('/classes/trainer');
export const createClass = (data) => api.post('/classes', data);
export const createClassAdmin = (data) => api.post('/classes/admin', data);
export const updateClass = (id, data) => api.put(`/classes/${id}`, data);
export const deleteClass = (id) => api.delete(`/classes/${id}`);
export const getAllClasses = () => api.get('/classes');
export const getClassById = (id) => api.get('/classes/' + id);



export const getClassesByTrainerUserId = (trainerUserId) => {
  return api.get(`/classes/trainer/${trainerUserId}`);
};






export const getUpcomingClasses = () => api.get('/classes/upcoming');

File: backend/src/main/java/com/example/Elitfit/Service/ClassService.java
package com.example.Elitfit.Service;

import com.example.Elitfit.Dto.ClassDto;
import java.util.List;

public interface ClassService {
    // Add authenticatedUserId param here
    ClassDto createClass(ClassDto dto, Long authenticatedUserId);
    ClassDto createClassAdmin(ClassDto dto);
    List<ClassDto> getAllClasses();
    ClassDto getClassById(Long id);
    ClassDto updateClass(Long id, ClassDto dto);
    void deleteClass(Long id);
    List<ClassDto> getUpcomingClasses();
    List<ClassDto> getClassesByTrainerUserId(Long trainerUserId);
    List<ClassDto> getClassesByTrainerId(Long trainerId); // already exists


}
